Sources/RoutingKit/Parameters.swift

```diff
@@ -55,4 +57,16 @@ public struct Parameters {
     public mutating func set(_ name: String, to value: String?) {
         self.values[name] = value?.removingPercentEncoding
     }
+    
+    /// Store and fetch `catchall` matches in the bag.
+    ///
+    /// - note: The value will be percent-decoded.
+    public var catchall: [String] {
+        get {
+            self.catchallValues
+        }
+        set {
+            self.catchallValues = newValue.map { $0.removingPercentEncoding ?? $0 }
```

Comment:
Could we instead remove percent encoding when `catchall` is read? Or even better, lazily remove percent encoding the first time `catchall` is read after it's set. Right now, this PR is removing percent encoding on strings anytime a `.catchall` route is hit. That will be a fairly large performance hit that we can avoid in cases where the user never accesses the catchall values.